pyiron_atomistics/atomistics/structure/factory.py

```diff
@@ -407,3 +416,68 @@ def from_pymatgen(pymatgen_obj):
+            length_step (int): The length of the step along the step direction in atoms eg., 3
+            length_kink (int): The length of the kink along the kink direction in atoms eg., 1
+            layers (int): Number of layers of the high_index_surface eg., 60
+            vacuum (float): Thickness of vacuum on the top of the slab
+
+        Returns:
+            high_index_surface: The high miller index surface which can be used to create slabs
+            fin_kink_orientation: The kink orientation lying in the terrace
+            fin_step_orientation: The step orientation lying in the terrace
+            slab: pyiron_atomistics.atomistics.structure.atoms.Atoms instance Required surface
+        """
+        basis = self.crystal(element=element, bravais_basis=crystal_structure, lattice_constant=lattice_constant)
+        sym = basis.get_symmetry()
+        eqvdirs = np.unique(np.matmul(sym.rotations[:], (np.array(step_orientation))), axis=0)
+        eqvdirk = np.unique(np.matmul(sym.rotations[:], (np.array(kink_orientation))), axis=0)
+        eqvdirs_ind = np.where(np.dot(np.squeeze(eqvdirs), terrace_orientation) == 0)[0]
+        eqvdirk_ind = np.where(np.dot(np.squeeze(eqvdirk), terrace_orientation) == 0)[0]
+        if len(eqvdirs_ind) == 0:
+            raise ValueError('Step orientation vector should lie in terrace.\
+            For the given choice I could not find any symmetrically equivalent vector that lies in the terrace.\
+            please change the stepOrientation and try again')
+        if len(eqvdirk_ind) == 0:
+            raise ValueError('Kink orientation vector should lie in terrace.\
+            For the given choice I could not find any symmetrically equivalent vector that lies in the terrace.\
+            please change the kinkOrientation and try again')
+        temp = (np.cross(np.squeeze(eqvdirk[eqvdirk_ind[0]]), np.squeeze(eqvdirs))).tolist().index(terrace_orientation)
+        fin_kink_orientation = eqvdirk[eqvdirk_ind[0]]
+        fin_step_orientation = eqvdirs[temp]
+        vec1 = (np.asanyarray(fin_step_orientation).dot(length_step)) + \
+               (np.asanyarray(fin_kink_orientation).dot(length_kink))
+        vec2 = (np.asanyarray(fin_kink_orientation).dot(length_terrace)) + step_down_vector
+        high_index_surface = np.cross(np.asanyarray(vec1), np.asanyarray(vec2))
+        high_index_surface = np.array(high_index_surface / np.gcd.reduce(high_index_surface), dtype=int)
+        surf = ase_surf(basis, high_index_surface, layers, vacuum)
+        sga = SpacegroupAnalyzer(pyiron_to_pymatgen(ase_to_pyiron(surf)))
+        pmg_refined = sga.get_refined_structure()
+        slab = pymatgen_to_pyiron(pmg_refined)
+        slab.positions[:, 2] = slab.positions[:, 2] - np.min(slab.positions[:, 2])
+        slab.set_pbc = True
+        slab.set_initial_magnetic_moments(np.repeat(1.0, len(slab)))
```

Comment:
Is this line necessary? Why set initial magnetic moments to one?

Reply:
For SPHInX jobs, if I do not initialize the magnetic moments to some initial values, it's not considering spins for calculations. That's why I have been initializing them to 1. Let me know if there is another way.

Reply:
> 
> 
> For SPHInX jobs, if I do not initialize the magnetic moments to some initial values, it's not considering spins for calculations. That's why I have been initializing them to 1. Let me know if there is another way.

But since this routine will be used to set up high index surfaces in general for both spin/non-spin polarized as well as non-electronic structure calculations, this line should be removed and should only be used when setting up sphinx calculations with magnetic moments.